persistent/Database/Persist/Types/Base.hs

```diff
@@ -461,7 +466,7 @@ data SqlType = SqlString
              | SqlOther T.Text -- ^ a backend-specific name
     deriving (Show, Read, Eq, Typeable, Ord)
 
-data PersistFilter = Eq | Ne | Gt | Lt | Ge | Le | In | NotIn
+data PersistFilter = Eq | Ne | Gt | Lt | Ge | Le | In | NotIn | Like | NotLike
```

Comment:
I feel like `Like` and `NotLIke` are really more appropriate as a `BackendSpecificFilter` since they're only really supported in SQL databases (even if some NoSQL databases can support them; eg Mongo can use regex).